### PR TITLE
(WIP) Node agent system detection and integration modules

### PR DESCRIPTION
**Pull request by brainfunked**
Against https://github.com/Tendrl/specifications/issues/87.

Currently details the node agent design wrt the system detection modules and the integration modules.

### CODE REVIEW

File: specs/node_agent_flows_and_namespaces.adoc [hunk 1]
```diff
@@ -0,0 +1,151 @@
+// vim: tw=79
+:toc:
```
Comment (shtripat) [inline specs/node_agent_flows_and_namespaces.adoc:2]: The only point I have at the moment is that it looks very high level to me and with more examples and snippets added from definitions would make it clearer for everybody else.

File: specs/node_agent_flows_and_namespaces.adoc [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+
+* By default, two types of integration modules need to be shipped with the node
+  agent:
+** Service detection modules, which have been covered in the <<Service Detection Modules>> section.
+** Provisioning modules to deploy the integration bridges or provisioning
+   systems (gdeploy, ceph-ansible etc.) for a specific storage system, which
+   have been covered in the <<Provisioning Modules>> section.
+* For both types of modules, node agent needs to define a separate directory
+  from where the specific type of integration provisioning modules can be loaded.
+* The modules can be python classes with a specific interface method to be
```
Comment (shtripat) [inline specs/node_agent_flows_and_namespaces.adoc:98]: I feel this spec should list wire-frame of these interfaces in implementation section.

File: specs/node_agent_flows_and_namespaces.adoc [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+  https://github.com/Tendrl/specifications/issues/49[#49].
+* Various provisioning flows,
+  https://github.com/Tendrl/specifications/issues/47[#47],
+  https://github.com/Tendrl/specifications/issues/51[#51],
+  https://github.com/Tendrl/specifications/issues/52[#52].
+
+
+== Proposed Changes
+
+=== Platform Detection Modules
```
Comment (nnDarshan) [inline specs/node_agent_flows_and_namespaces.adoc:66]: will there be multiple modules to detect platform ? cant this be done by a single module if its just detection of the OS and its version ?

File: specs/node_agent_flows_and_namespaces.adoc [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+            # Provide either a list of success or failure exit codes. Anything
+            # else would indicate the other condition.
+            success_exit_code: Array
+      module:
+        # A specific module in the code
+        parameters:
+          - module: Module
+            parameters: Hash
+    storage_system_types:
+      ceph: namespace.tendrl.integrations.ceph
```
Comment (r0h4n) [inline specs/node_agent_flows_and_namespaces.adoc:248]: we call it "namespace.tendrl.ceph_integration", i suppose we have to change it to "namespace.tendrl.integrations.ceph"

same for gluster

**Review by r0h4n — CHANGES_REQUESTED** [review]

**Pull request merged**